Realm/RLMObjectBase.mm

```diff
@@ -441,3 +444,30 @@ + (NSArray *)requiredPropertiesForClass:(Class)cls {
 }
 
 @end
+
+@interface RLMObjectBase (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMObjectBase (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    // FIXME: Once merged, should be easier to get schema ref (https://github.com/realm/realm-cocoa/pull/3941)
+    auto className = RLMStringDataWithNSString(self.objectSchema.className);
+    auto& schema = _realm->_realm->schema();
+    auto objectSchema = schema.find(std::move(className));
+    REALM_ASSERT(objectSchema != schema.end());
+    return AnyThreadConfined(Object(_realm->_realm, *objectSchema, _row));
+}
+
+- (NSNull *)rlm_handoverMetadata {
+    return [NSNull null];
```

Comment:
It doesn't seem that odd to me. The type of `rlm_handoverMetadata` is `() -> Any`, so this function must return a non-nil value. I could change the protocol such that this return value is nullable, but that clearly better to me.